### ISSUE TITLE
Defining Arrays / Lists rather than string objects

### ISSUE THREAD

**Issue by mittalkunal** [Netcentric/accesscontroltool#184]
Hi

Is it possible to have support for Defining variables which can have multiple values like Array / List.

We have a use case where we need to create groups and assign permissions for mutiple hierarchies of content but not all. What I was thinking was to define a variable at global level and use the same for creation of groups as well as assigning permissions. Something like below:

```
- DEF markets=[us,ca]

- group_config:

  - FOR market in ${markets}:

    - editors-${market}:
       - name: Editor for ${market}
         isMemberOf: 
         path: /home/groups/${market}

- ace_config:

  - FOR market in ${markets}:

    - editor-{market}:

      - path: /content/xyz/${market}
        permission: allow
        actions: read,modify,create,delete,acl_read,acl_edit,replicate
        repGlob: */jcr:content*
```

Secondly if - IF ${endsWith(site.name,'-master')}: can have an array as well so that we can use the if condition to work with various values rather than 1

**Comment by ghenzler** [comment]
@mittalkunal See #194 to make your example work. 

> Secondly if - IF ${endsWith(site.name,'-master')}: can have an array as well so that we can use the if condition to work with various values rather than 1

I'm not sure about this, once you can loop, the if should be in the loop and then you can call the `endsWith()` with the scalar value?

**Comment by ghenzler** [comment]
Please open a new issue for the special IF syntax if needed. 

The main part of this ticket (allowing to use array variables for FOR loops) is fixed with now merged PR #194

**Comment by ghenzler** [comment]
Fixed in v2.0.0: https://github.com/Netcentric/accesscontroltool/releases/tag/2.0.0

**Comment by mittalkunal** [comment]
:+1: Thanks a lot @ghenzler : Appreciate that it is added in the latest release. Will check the array thing and raise a new ticket if required.
